Remove references to the <Markdown> component (#1045)
withastro/docs: src/pages/en/migrate.md

To make migrating easier, a new [legacy flag](/en/reference/configuration-reference/#legacyastroflavoredmarkdown) can be used to re-enable previous Markdown features.

### `<Markdown />` Component Removed

> If you were previously using Astro’s built-in `<Markdown />` component, this has been moved to a separate package. You will now need to install `@astrojs/markdown` and update your imports accordingly. For more details, see [the `@astrojs/markdown` README](https://github.com/withastro/astro/tree/main/packages/markdown/component).

Reviewer: Thanks @matthewp ! 

@delucis , I'm assuming that this `@astrojs/markdown` component does NOT get automatically added to our Integrations page?

- Do we *want* a legacy package on our integrations page, so that we don't *need* to send people externally to the README but can instead have a page in Docs? Or, do we specifically NOT want this showing up in Docs, because we don't want people using it unless they really have to?

- Do we *want* to take this opportunity here in the migration guide to encourage people to transition to MDX, or simply say, "Here's where the new package is?"

Author: I think we don't want people using it unless they really have to.

Agree that it's a good opportunity to push people towards MDX, I'll add that.

To make migrating easier, a new [legacy flag](/en/reference/configuration-reference/#legacyastroflavoredmarkdown) can be used to re-enable previous Markdown features.

### `<Markdown />` Component Removed

> If you were previously using Astro's built-in `<Markdown />` component, this has been moved to a separate package. You will now need to install `@astrojs/markdown` and update your imports accordingly. For more details, see [the `@astrojs/markdown` README](https://github.com/withastro/astro/tree/main/packages/markdown/component).

Reviewer: ```suggestion
Astro's built-in `<Markdown />` component has been moved to a separate package. To continue using this component, you will now need to install `@astrojs/markdown` and update your imports accordingly. For more details, see [the `@astrojs/markdown` README](https://github.com/withastro/astro/tree/main/packages/markdown/component).
```

Just a little more direct, if you like.

> Astro now has support for [MDX](https://mdxjs.com/) through our [MDX integration](https://github.com/withastro/astro/tree/main/packages/integrations/mdx). MDX gives you the ability to create components that can be imported into an Astro page that include both Markdown and component usage. MDX make a good alternative for the Markdown component due to it's large community and stable APIs.

Reviewer: ```suggestion
:::tip
Astro now has support for [MDX](https://mdxjs.com/) through our [MDX integration](https://github.com/withastro/astro/tree/main/packages/integrations/mdx). MDX gives you the ability to include both Markdown and imported components in the same file. MDX can be good alternative for the `<Markdown />` component due to its large community and stable APIs.
:::
```

Wondering whether this might be a good "aside"?

I did also condense this a little for readablility, which admittedly loses a bit of the nuance but might be good enough for a note in the migration guide?
